Yoast/YoastSEO.js
# Change isApplicable for topic density and topic distribution

## Summary

This PR can be summarized in the following changelog entry:

* Adds link to the knowledge-base article about keyword density and keyword distribution to the respective assessments

## Relevant technical choices:

* Removes old synonym definition from isApplicable, which relied on the presence of comma-separated list in the focus keyword field.
* Adds new synonym definition to isApplicable, which relies on paper.hasSynonyms().
* Improves markings on keywords and synonyms so that word boundaries are not highlighted.

## Test instructions

This PR can be tested by following these steps:

* Check if keywordDistribution and keywordDensity are triggered if no synonyms are supplied.
* Check if topicDistribution and topicDensity are triggered if synonyms are supplied.
* Check if none of the two is triggered if no keyword is supplied.
* Check if links in the feedback strings are working with and without synonyms.
* Check if markings work well and do not grab the word boundary along (e.g., for `>keyword` only `keyword` should be marked)

Fixes #1565
